File: src/VocaNova.Mobile/lib/features/dictionary/domain/models/word_detail.dart
class WordDetail {
  const WordDetail({
    required this.wordId,
    required this.word,
    required this.isPhrase,
    required this.senses,
    required this.examples,
    required this.relations,
    required this.audio,
    required this.topics,
    this.cefr,
    this.phoneticUk,
    this.phoneticUs,
    this.imageUrl,
  });

  final int wordId;
  final String word;
  final bool isPhrase;
  final String? cefr;
  final String? phoneticUk;
  final String? phoneticUs;
  final String? imageUrl;
  final List<WordSense> senses;
  final List<WordExample> examples;
  final List<WordRelation> relations;
  final List<WordAudio> audio;
  final List<WordTopic> topics;
}

class WordSense {
  const WordSense({
    required this.senseId,
    required this.order,
    required this.wordClass,
    required this.englishDefinition,
    required this.examples,
    required this.relations,
    this.vietnameseMeaning,
  });

  final int senseId;
  final int order;
  final String wordClass;
  final String englishDefinition;
  final String? vietnameseMeaning;
  final List<WordExample> examples;
  final List<WordRelation> relations;
}

class WordExample {
  const WordExample({required this.exampleEn, this.exampleVi});

  final String exampleEn;
  final String? exampleVi;
}

class WordRelation {
  const WordRelation({
    required this.relationType,
    required this.relatedWord,
    this.linkedWordId,
  });

  final String relationType;
  final String relatedWord;
  final int? linkedWordId;
}

class WordAudio {
  const WordAudio({required this.accent, required this.url});

  final String accent;
  final String url;
}

class WordTopic {
  const WordTopic({
    required this.topicId,
    required this.name,
    this.nameVi,
    this.icon,
  });

  final int topicId;
  final String name;
  final String? nameVi;
  final String? icon;
  String get displayName => nameVi?.trim().isNotEmpty == true ? nameVi! : name;
}

class UserListSummary {
  const UserListSummary({
    required this.listId,
    required this.listName,
    required this.wordCount,
  });

  final int listId;
  final String listName;
  final int wordCount;
}

File: src/VocaNova.API/Features/Dictionary/BLL/Models/WordDetail.cs
namespace VocaNova.API.Features.Dictionary.BLL.Models;

public sealed record WordDetail(
    uint WordId,
    string Word,
    string WordKey,
    string? Cefr,
    string? PhoneticUk,
    string? PhoneticUs,
    string? ImageUrl,
    bool IsPhrase,
    IReadOnlyCollection<WordSense> Senses,
    IReadOnlyCollection<WordExample> Examples,
    IReadOnlyCollection<WordRelation> Relations,
    IReadOnlyCollection<WordAudio> Audio,
    IReadOnlyCollection<WordDerivedForm> DerivedForms,
    IReadOnlyCollection<WordIdiom> Idioms,
    IReadOnlyCollection<WordTopic> Topics,
    string Status,
    DateTime CreatedAt,
    DateTime UpdatedAt);

public sealed record WordSense(
    uint SenseId,
    int Order,
    string WordClass,
    string EnglishDefinition,
    string? VietnameseMeaning,
    IReadOnlyCollection<WordExample> Examples,
    IReadOnlyCollection<WordRelation> Relations);

public sealed record WordExample(
    uint ExampleId,
    uint? SenseId,
    string ExampleEn,
    string? ExampleVi,
    int Order);

public sealed record WordRelation(
    uint RelationId,
    uint? SenseId,
    string RelationType,
    string RelatedWord,
    uint? LinkedWordId,
    bool IsQuizEligible);

public sealed record WordAudio(
    uint AudioId,
    string Accent,
    string Source,
    string Url,
    string Status);

public sealed record WordDerivedForm(
    uint DerivedId,
    string DerivedWord,
    uint? LinkedWordId,
    string? WordClass);

public sealed record WordIdiom(
    uint IdiomId,
    string IdiomText,
    string? MeaningEn,
    string? MeaningVi);

public sealed record WordTopic(
    uint TopicId,
    string Name,
    string? NameVi,
    string? Icon);
